internal/provider/common.go

```diff
@@ -70,17 +86,33 @@ func checkPlatformWrongParamsInput(platform *models.Platform, userManagedNetwork
 }
 
 func doesPlatformAllowUMNOrNone(platform *models.Platform, cluster *common.Cluster) bool {
-	if platform != nil && isUMNAllowedForPlatform(platform) && !isPlatformNone(platform) {
+	if platform != nil && isUMNAllowedForPlatform(platform) && !isUMNMandatoryForPlatform(platform) {
 		return true
 	}
 
-	if platform == nil && isUMNAllowedForCluster(cluster) && !isClusterPlatformNone(cluster) {
+	if platform == nil && isUMNAllowedForCluster(cluster) && !isUMNMandatoryForCluster(cluster) {
 		return true
 	}
 
 	return false
 }
 
+func createPlatformFromType(platformType models.PlatformType) *models.Platform {
+	platform := &models.Platform{
+		Type: &platformType,
+	}
+	return updatePlatform(platform)
+}
+
+func updatePlatform(platform *models.Platform) *models.Platform {
```

Comment:
maybe give it a better name? I couldn't understand from the its name what it's doing

Reply:
renamed it `updatePlatformIsExternal`